### PR TITLE
Update token revocation

### LINKED ISSUE #29: JWT revocation doc should be restructured
### Description:

This doc should be divided in to Concepts and How to
### Content Positioning in Documentation:

- Link:
https://mg.docs.wso2.com/en/latest/how-tos/security/jwt-token-revocation/#jwt-token-revocation

- Heading (& Step):

- Any other reference:

### PR DESCRIPTION
**Pull request by praminda**
### Purpose

Restructure and update contents or token revocation doc.
* Page is renamed to match both jwt and opaque revocation content
* Default persistent storage is changed to eventhub
* etcd curl commands are migrated to `etcdctl`

### Issues

Fixes #29

---
#### Maintainers: Check before merge
- [x] Manual verification

### CODE REVIEW

File: en/docs/concepts/revoked-tokens.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Revoked Tokens
+
+There are two usecase where API Microgateway needs to know when a token is revoked by the Security Token Service (STS).
```
Comment (shaniR) [inline en/docs/concepts/revoked-tokens.md:3]: The API Microgateway is notified when a token is revoked by the Security Token Service (STS) as described in the following two instances.

File: en/docs/concepts/revoked-tokens.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+#### Persistent Notifier
+
+When using Persistent Notifications, WSO2 API Microgateway uses a persistent storage mechanism to link the Security Token Service (STS) and the WSO2 API Microgateway servers. Whenever a token revoke request is received, the STS publishes a message to the persistent storage. When a new WSO2 API Microgateway server spins up, it pulls the list of revoked tokens from the persistent storage, and stores them in the revoked jti (JWT ID) cache. The latter mentioned process only takes place once, and the state of the token is preserved. The state of the revoked token is used at restarts and when new API Microgateways join the cluster of Microgateway servers. By default, WSO2 API Microgateway can use WSO2 API Manager or [etcd](https://github.com/etcd-io/etcd) servers as its persistent storage when working with persistent notifications. However, unlike when using the real-time notifications, you can use any persistent storage and a custom implementation.
+
+For more information on how to enable and work with this feature, check [How to revoke tokens](/how-tos/security/rejecting-revoked-tokens)
```
Comment (AmaliMatharaarachchi) [inline en/docs/concepts/revoked-tokens.md:25]: Shall we make it a note/info?

Comment (praminda) [reply en/docs/concepts/revoked-tokens.md:25]: This is the main link from concept to howto. So better keep it in the main structure.

File: en/docs/how-tos/security/rejecting-revoked-tokens.md [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+      </td>
+      <td>
+      Enter the password of your persistent storage server (e.g., etcd server). The default value for the etcd server is "root".
+      </td>
+      </tr>
+      </tbody>
+    </table>
+
+    !!! note
+        When working with persistent notifications, if you introduced new configurations in your custom implementation, then you have to provide the corresponding values for those propertyies in the `<MGW_HOME>/conf/micro-gw.conf` file under the `[tokenRevocationConfig.persistent]` section.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/security/rejecting-revoked-tokens.md:176]:
```suggestion
        When working with persistent notifications, if you introduced new configurations in your custom implementation, then you have to provide the corresponding values for those properties in the `<MGW_HOME>/conf/micro-gw.conf` file under the `[tokenRevocationConfig.persistent]` section.
```

File: en/docs/how-tos/security/rejecting-revoked-tokens.md [hunk 4]
```diff
@@ -0,0 +1,453 @@
+# Rejecting Revoked Tokens
+Follow below steps to enable Real-time and Persistent token revocation. This documentation explains steps required to handle revoked JWTs. Same steps can be used for handling revoked opaque tokens. However for opaque tokens, you only need to enable realtime notifications.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/security/rejecting-revoked-tokens.md:2]:
```suggestion
Follow the below steps to enable Real-time and Persistent token revocation. This documentation explains the steps required to handle revoked JWTs. The same steps can be used for handling revoked opaque tokens. However, for opaque tokens, you only need to enable real-time notifications.
```

File: en/docs/how-tos/security/rejecting-revoked-tokens.md [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+
+5.  Establish security between the persistent storage (etcd server), WSO2 API Microgateway, and the STS.
+
+    !!! note
+        This is only required if you are working in a production environment.
+
+    1.  Create a self-signed certificate for the persistent storage.
+    2.  Add the self-signed certificate to the Ballerina Trust Store ( `ballerinaTruststore.p12` ) in the WSO2 API Microgateway project and to the Client Trust Store ( `client-truststore.jks` ) in the STS (e.g., WSO2 API Manager).
+
+6.  Create and publish and API.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/security/rejecting-revoked-tokens.md:314]:
```suggestion
6.  Create and publish an API.
```

File: en/docs/how-tos/security/rejecting-revoked-tokens.md [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+
+    !!! note
+        This step is only required if you do not already have an existing API.
+
+    For this example, let's skip this step and use the [Petstore OpenAPI](https://github.com/wso2/product-microgateway/blob/master/samples/petstore_basic.yaml) .
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/security/rejecting-revoked-tokens.md:319]:
```suggestion
    For this example, let's skip this step and use the [Petstore OpenAPI](https://github.com/wso2/product-microgateway/blob/master/samples/petstore_basic.yaml).
```

File: en/docs/how-tos/security/rejecting-revoked-tokens.md [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+    ```
+
+    ``` java tab="Sample Response"
+    Project 'petstore' is initialized successfully.
+    ```
+
+8.  Build the WSO2 API Microgateway project (e.g., petstore).
+
+    1.  Add an API or multiple APIs to the project.
+        Navigate to the `/petstore/api_definitions` directory. Add the API definition(s) to this  directory. For this example, let's add the [petstore](https://github.com/wso2/product-microgateway/blob/master/samples/petstore_basic.yaml) open API definition.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/security/rejecting-revoked-tokens.md:338]:
```suggestion
        Navigate to the `/petstore/api_definitions` directory. Add the API definition(s) to this directory. For this example, let's add the [petstore](https://github.com/wso2/product-microgateway/blob/master/samples/petstore_basic.yaml) open API definition.
```

File: en/docs/how-tos/security/rejecting-revoked-tokens.md [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+    INFO  [wso2/gateway/src/gateway/utils] - [APIGatewayListener] [-] HTTPS listener is active on port 9095
+    INFO  [wso2/gateway/src/gateway/utils] - [RevokedJwtRetrievalTask] [-] Revoked jwt retrieval successful. Stopping the timer task ...
+    INFO  [wso2/gateway/src/gateway/utils] - [RevokedJwtRetrievalTask] [-] Revoked jwt retrieval task stopped successfully
+    ```
+
+10. Test the synchronization process of token revocation.
+
+    1. Send the revoke request using the extracted jti as the token.
+
+        ``` java tab="Format"
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/security/rejecting-revoked-tokens.md:398]:
```suggestion
        ``` text tab="Format"
```

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**